install/upgrade.go

```diff
@@ -81,7 +88,55 @@ func (k *K8sInstaller) Upgrade(ctx context.Context) error {
 		patched++
 	}
 
-	if patched > 0 && k.params.Wait {
+	hubbleDeployment, err := k.client.GetDeployment(ctx, k.params.Namespace, defaults.RelayDeploymentName, metav1.GetOptions{})
+	if err != nil {
+		k.Log("ℹ️  Unable to retrieve Deployment of hubble-relay")
+		message++
+	}
+
+	message, err = k.ValidateHubble(ctx)
+	if err != nil {
+		k.Log("Error:%q", err)
+	}
+
+	if message >= 1 && err == nil {
```

Comment:
Rather than accumulating state in local variables (e.g. counting `message`s and populating `err`) and then trying to work out what happened from the state (as is done here), I'd suggest structuring this a clear sequence of steps, each of which does its own error handling.